src/otx/core/metrics/fmeasure.py

```diff
@@ -643,22 +643,31 @@ class FMeasure(Metric):
 
     def __init__(
         self,
-        num_classes: int,
-        best_confidence_threshold: float | None = None,
+        label_info: LabelInfo,
         vary_nms_threshold: bool = False,
         cross_class_nms: bool = False,
     ):
         super().__init__()
         self.vary_nms_threshold = vary_nms_threshold
         self.cross_class_nms = cross_class_nms
-        self.preds: list[list[tuple]] = []
-        self.targets: list[list[tuple]] = []
-        self.num_classes: int = num_classes
+        self.label_info: LabelInfo = label_info
 
         self._f_measure_per_confidence: dict | None = None
         self._f_measure_per_nms: dict | None = None
-        self._best_confidence_threshold: float | None = best_confidence_threshold
+        self._best_confidence_threshold: float | None = None
         self._best_nms_threshold: float | None = None
+        self._f_measure = 0.0
```

Comment:
I think we can't set pre-defined confidence threshold during training phase. In Geti, user can specify confidence threshold they want, and it is expected to validate using that confidence threshold during training.

Reply:
I think that there was not clear interface about that you mentioned too before this PR. If you think it is necessary, add it to the constructor. But, I don't think that it is necessary.